File: decrypt_admin.js
const crypto = require('crypto');
const fs = require('fs');

const password = '@thzyvxkupka3453';
const html = fs.readFileSync('adminpanelacess/index.html', 'utf8');
const payloadMatch = html.match(/<div id="encryptedPayload" style="display:none;">\s*([\s\S]*?)\s*<\/div>/);
const payloadB64 = payloadMatch[1].trim();
const payloadJson = JSON.parse(Buffer.from(payloadB64, 'base64').toString('utf8'));

const salt = Buffer.from(payloadJson.salt, 'base64');
const iv = Buffer.from(payloadJson.iv, 'base64');
const ciphertext = Buffer.from(payloadJson.data, 'base64');

crypto.pbkdf2(password, salt, 100000, 32, 'sha256', (err, key) => {
    if (err) throw err;
    const decipher = crypto.createDecipheriv('aes-256-gcm', iv, key);
    const tag = ciphertext.slice(ciphertext.length - 16);
    const encryptedContent = ciphertext.slice(0, ciphertext.length - 16);

    decipher.setAuthTag(tag);
    try {
        let decrypted = decipher.update(encryptedContent, null, 'utf8');
        decrypted += decipher.final('utf8');
        fs.writeFileSync('admin_unencrypted.html', decrypted);
        console.log("Decrypted successfully!");
    } catch (e) {
        console.error("error decrypting: ", e);
    }
});

File: decrypt_admin.py
import base64
import json
import re

try:
    from Crypto.Cipher import AES
    from Crypto.Protocol.KDF import PBKDF2
    from Crypto.Hash import SHA256
except ImportError:
    import sys
    print("pycryptodome not installed")
    sys.exit(1)

password = b'@thzyvxkupka3453'

with open('adminpanelacess/index.html', 'r', encoding='utf8') as f:
    html = f.read()

m = re.search(r'<div id="encryptedPayload" style="display:none;">\s*([\s\S]*?)\s*</div>', html)
payload_b64 = m.group(1).strip()
payload = json.loads(base64.b64decode(payload_b64).decode('utf8'))

salt = base64.b64decode(payload['salt'])
iv = base64.b64decode(payload['iv'])
ciphertext = base64.b64decode(payload['data'])

# The tag is the last 16 bytes
tag = ciphertext[-16:]
data = ciphertext[:-16]

key = PBKDF2(password, salt, 32, count=100000, hmac_hash_module=SHA256)

cipher = AES.new(key, AES.MODE_GCM, nonce=iv)
decrypted = cipher.decrypt_and_verify(data, tag)

with open('admin_unencrypted.html', 'w', encoding='utf8') as f:
    f.write(decrypted.decode('utf8'))
print("Decrypted successfully")
